torchtune/datasets/_sft.py

```diff
@@ -117,7 +128,17 @@ def __getitem__(self, index: int) -> Dict[str, Any]:
         sample = self._data[index]
         return self._prepare_sample(sample)
 
-    def _prepare_sample(self, sample: Mapping[str, Any]) -> Dict[str, Any]:
+
+class PrepareSample:
```

Comment:
Refactoring this into it's own Callable class that can be used by both the current torch.utils.Dataset and the torchdata.nodes